examples/pipelines/fl_cross_silo_factory/submit.py

```diff
@@ -282,132 +286,71 @@ def silo_scatter_subgraph(
-    # scatter is our subgraph above
+pipeline_job = scatter_gather(
     scatter=silo_scatter_subgraph,
-    # gather is directly a command component
-    gather=aggregate_component,
-    # a function to map name of outputs from each scatter into a name for the inputs of gather
-    scatter_to_gather_map=lambda output_name, silo_index: f"input_silo_{silo_index+1}",
-    # a function to map name of outputs of gather to inputs of scatter
-    gather_to_accumulator_map=lambda output_name: "aggregated_checkpoint",
-    # a dictionary describing the "accumulator"
-    # i.e. the input that is passed from gather to scatter in the next iteration
-    accumulator={
-        # this key needs to be the name of the input expected by scatter
-        # factory will use gather_to_accumulator_map(key) to map
-        # to the input name required by scatter
-        "name": "aggregated_checkpoint",
-        # we can provide an initial input for the first iteration
-        # but this is an optional input
-        "initial_input": None,
-    },
-    # how many iterations
-    iterations=YAML_CONFIG.training_parameters.num_of_iterations,
-    # any additional kwarg is considered constant and given to scatter as is
-    lr=YAML_CONFIG.training_parameters.lr,
-    batch_size=YAML_CONFIG.training_parameters.batch_size,
-    epochs=YAML_CONFIG.training_parameters.epochs,
-    # Differential Privacy
-    dp=YAML_CONFIG.training_parameters.dp,
-    # DP target epsilon
-    dp_target_epsilon=YAML_CONFIG.training_parameters.dp_target_epsilon,
-    # DP target delta
-    dp_target_delta=YAML_CONFIG.training_parameters.dp_target_delta,
-    # DP max gradient norm
-    dp_max_grad_norm=YAML_CONFIG.training_parameters.dp_max_grad_norm,
-    # Total num of iterations
-    total_num_of_iterations=YAML_CONFIG.training_parameters.num_of_iterations,
+    gather=gather_pipeline,
+    scatter_strategy=scatter_configs,
+    gather_strategy=gather_config,
+    scatter_to_gather_map=lambda _, silo_index: f"input_silo_{silo_index}",
+    gather_to_scatter_map=lambda _: "checkpoint",
+    iterations=YAML_CONFIG.iterations,
+    scatter_constant_inputs=scatter_constant_inputs,
 )
 
-# 4. Validate the pipeline using soft rules
-
-print(pipeline_job)  # print yaml for visual debugging
+#########################################
+### E. Pipeline Validation (optional) ###
+#########################################
 
-# use a default set of rules
-builder.set_default_affinity_map()
-
-# run affinity map validation
-print("Running soft validation...")
-builder.soft_validate(
-    pipeline_job,
-    raise_exception=not (
-        args.ignore_validation
-    ),  # set to False if you know what you're doing
-)
+if not args.ignore_validation:
+    from fl_helper import FLValidationEngine
 
+    fl_val_engine = FLValidationEngine(scatter_configs, gather_config)
+    fl_val_engine.soft_validate(pipeline_job)
```

Comment:
instead of running soft_validate() only if not ignore_validation, we could use soft_validate's argument raise_exception=True/False to ALWAYS do the validation, but not except if ignored

Reply:
done